examples/fishnet_api/main.py

```diff
@@ -286,140 +348,150 @@ async def request_execution(
-                if data.id_hash in ds_ids:
-                    ds_ids.append(data.id_hash)
-
-            executions = await Execution.fetch(ds_ids)
-            if executions:
-                for execution in executions:
-                    if ds_ids and execution.datasetID in ds_ids:
-                        execution.status = ExecutionStatus.PENDING
-                        requests.append(rec.upsert())
-                await asyncio.gather(requests)
-                return {"Success": "Permissions Approved "}
-            else:
-                return {"Execution": "No Execution found "}
-        else:
-            return {"Dataset": "No Dataset found"}
-    else:
-        return {"Permission": "No Permission Found with this Hashes"}
+    await asyncio.gather(*requests)
+    return permission_records
 
 
 @app.put("/permissions/deny")
-async def deny_permissions(permission_hashes: List[str]) -> DenyPermissionsResponse:
+async def deny_permissions(permission_hashes: List[str]) -> List[Permission]:
     """
-    Deny a list of permissions by their item hashes.
+    Deny permission.
+    This EndPoint will approve a list of permissions by their item hashes
+    If an `id_hashes` is provided, it will change all the Permission status
+    to DENIED.
     """
-    permission_record = await Permission.fetch(permission_hashes)
-    if permission_record:
-        ts_ids = []
-        requests = []
-        for permission in permission_record:
-            # deny permissions and update records
-            permission.status = PermissionStatus.DENIED
-            ts_ids.append(permission.timeseriesID)
-            requests.append(permission.upsert())
-        dataset = await Dataset.fetch(ts_ids)
-        ds_ids = []
-        if dataset:
-            for data in dataset:
-                ds_ids.append(data.id_hash)
-            # Avoided fetching all executions
-            execution = await Execution.fetch(ds_ids)
-            if execution:
-                for rec in execution:
-                    # get all executions that are waiting for this permission(status == PENDING) and update their status to DENIED
-                    if rec.datasetID in ds_ids and rec.status == ExecutionStatus.PENDING:
-                        rec.status = ExecutionStatus.DENIED
-                        requests.append(rec.upsert())
-                # parellel processed
-                await asyncio.gather(requests)
-                return DenyPermissionsResponse(success=True, message="Denied all Permissions")
-            else:
-                return DenyPermissionsResponse(success=True, message="No Execution found")
-        else:
-            return {"message": "No Timeseries found"}
-    else:
-        return {"message": "No Permission found with this Hashes"}
+    permission_records = await Permission.fetch(permission_hashes)
+    if not permission_records:
+        raise HTTPException(status_code=424, detail="No Permission found with this Hashes")
```

Comment:
404 would be more suitable here 
```suggestion
        raise HTTPException(status_code=404, detail="No Permission found with this Hashes")
```